src/hooks/coreSDK/useInitSafeCoreSDK.ts

```diff
@@ -23,7 +29,7 @@ export const useInitSafeCoreSDK = (): Error | null => {
         setSafeSDK(undefined)
         setError(e)
       })
-  }, [chain, wallet, safe, safeLoaded])
+  }, [chainId, wallet?.provider, wallet?.chainId, safe, safeLoaded])
```

Comment:
Narrowing the `wallet` dependency here also reduced the number of Core SDK calls from 2 to 1 when reloading the page normally.